lib/connectors/memory.js

```diff
@@ -104,45 +133,93 @@ Memory.prototype.collectionSeq = function(model, val) {
   return this.ids[model];
 };
 
+/**
+ * Create a queue to serialize file read/write operations
+ * @returns {*} The file operation queue
+ */
+Memory.prototype.setupFileQueue = function() {
+  var self = this;
+  if (!this.fileQueue) {
+    // Create a queue for writes
+    this.fileQueue = async.queue(function(task, cb) {
+      var callback = task.callback;
+      var file = self.settings.file;
+      if (task.operation === 'write') {
+        // Flush out the models/ids
+        var data = JSON.stringify({
+          ids: self.ids,
+          models: self.cache,
+        }, null, '  ');
+        debug('Writing cache to %s: %s', file, data);
+        fs.writeFile(file, data, function(err) {
+          debug('Cache has been written to %s', file);
+          cb(err);
+          callback && callback(err, task.data);
+        });
```

Comment:
The read/write requires/shares quite a bit context. I'm leaving the function as-is for simpler code.